server/api.js

```diff
@@ -110,21 +110,67 @@ router.post("/energizers", async (req, res) => {
 	}
 });
 
+router.post("/energizers/:id/rate", async (req, res) => {
+	const energizerId = req.params.id;
+	const rating = req.body.value;
+
+	try {
+		// Insert or update rating in the energizer_ratings table
+		await db.query(
+			`INSERT INTO energizer_ratings (energizer_id, rating) 
+				VALUES ($1, $2)`,
+			[energizerId, rating]
+		);
+
+		// Get the average rating of the energizer from the energizer_ratings table
+		const avgRatingResult = await db.query(
+			"SELECT ROUND(AVG(rating), 1) as average_rating FROM energizer_ratings WHERE energizer_id = $1",
+			[energizerId]
+		);
+		const averageRating = avgRatingResult.rows[0].average_rating;
+
+		// Update the average rating of the energizer in the energizers table
+		await db.query(
+			"UPDATE energizer_ratings SET average_rate = $1 WHERE id = $2",
+			[averageRating, energizerId]
+		);
+
+		// Get the updated energizer data
+		const energizerResult = await db.query(
+			`SELECT energizers.id, energizers.name, energizers.description, energizers.submission_date, ROUND(AVG(energizer_ratings.rating), 1) as rating 
+				FROM energizers 
+				LEFT JOIN energizer_ratings ON energizers.id = energizer_ratings.energizer_id
+    			GROUP BY energizers.id
+				WHERE energizers.id = $1`,
+			[energizerId]
+		);
+
+		const energizer = energizerResult.rows[0];
+
+		res.json({ energizer });
```

Comment:
Why is the entire energizer object being returned by this API endpoint? I would imagine the only thing that is necessary to return would be the new average?